client-course-schedulizer/src/components/App/App.tsx

```diff
@@ -13,6 +21,68 @@ export const App = () => {
   const [isCSVLoading, setIsCSVLoading] = useState(false);
   const { save } = useLocal("appState");
   const [appState, appDispatch] = useReducer(reducer(save), initialAppState);
+  const { schedule, fileUrl } = appState;
+
+  // Load a file from the GET parameters (either ?csv= or ?xlsx=),
+  // only do this once for each page refresh and only do this if the initial state is empty
+  // Ref: https://stackoverflow.com/questions/56663785/invalid-hook-call-hooks-can-only-be-called-inside-of-the-body-of-a-function-com
+  // Ref: https://www.gitmemory.com/issue/SheetJS/js-xlsx/1110/531003740
+  useEffect(() => {
+    const scheduleHasCourses = schedule.courses.length > 0;
+    if (!scheduleHasCourses) {
+      // eslint-disable-next-line no-restricted-globals
+      const url = location.href;
+      // TODO: We'll need to be more careful about this if we end up using more GET params
+      const csvIndex = url.indexOf("?csv=");
+      const xlsxIndex = url.indexOf("?xlsx=");
+      if (csvIndex !== -1) {
+        const csvUrl = url.slice(csvIndex + 5);
+        if (csvUrl !== "" && csvUrl !== fileUrl) {
+          setIsCSVLoading(true);
+          fetch(csvUrl)
+            .then((response) => {
+              if (response.status === 404) {
+                return null;
+              }
+              return response.text();
+            })
+            .then(async (result) => {
+              await appDispatch({ payload: { fileUrl: csvUrl }, type: "setFileUrl" });
```

Comment:
On line 40, we check against the `fileUrl` to make sure we aren't loading the schedule repeatedly if we have already loaded it once, right? That is why we set the `fileUrl` after checking `csvUrl !== fileUrl`

Reply:
Yep, otherwise every time the app is loaded, it loads the url's file (though maybe this problem could be solved if we redirect once the csv is loaded...)

Reply:
Okay, awesome!